### PR TITLE
Flink: Improve IcebergFilesCommitter logging

### PR DESCRIPTION
**Pull request by pvary**
Add some logging to the `IcebergFilesCommitter`. Otherwise it is hard to follow what happens when multiple commits are in progress.

### CODE REVIEW

File: flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java [hunk 1]
```diff
@@ -372,9 +382,10 @@ private void commitOperation(
     operation.commit(); // abort is automatically called if this fails.
     long durationMs = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNano);
     LOG.info(
-        "Committed {} to table {} in {} ms with summary: {}",
+        "Committed {} to table: {}, checkpointId {} in {} ms with summary: {}",
```
Comment (rdblue) [inline flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java:385]: Can we remove the summary? We don't need to log it twice.

**Review by rdblue — APPROVED** [review]
Overall looks good. I'd prefer to only log the summary once, if at all.

File: flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java [hunk 2]
```diff
@@ -361,7 +366,12 @@ private void commitOperation(
       String description,
       String newFlinkJobId,
       long checkpointId) {
-    LOG.info("Committing {} to table {} with summary: {}", description, table.name(), summary);
+    LOG.info(
+        "Committing {} to table: {}, checkpointId: {} with summary: {}",
```
Comment (stevenzwu) [inline flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java:370]: the log message doesn't read quite naturally. 

here is one alternative
```
LOG.info("Committing {} for checkpoint {} to table{} with summary: {}", description, checkpointId, table.name(), summary)
```

**Comment by pvary** [comment]
Merged to master.
Thanks for the review @rdblue and @stevenzwu!
